math/src/field/f64/mod.rs

```diff
@@ -569,3 +551,13 @@ fn mod_reduce(x: u128) -> u64 {
     let (result, over) = tmp0.overflowing_add(tmp1);
     result.wrapping_add(E * (over as u64))
 }
+
+/// Squares the base N number of times and multiplies the result by the tail value.
+#[inline(always)]
+fn exp_acc<const N: usize>(base: BaseElement, tail: BaseElement) -> BaseElement {
+    let mut result = base;
+    for _ in 0..N {
+        result = result.square();
+    }
+    result * tail
+}
```

Comment:
This is very nicely written and optimized.

I only see unit tests for 0 and 1.  Maybe I am missing something.  To know that this isn't a regression, maybe a unit test with 2 or some other larger number would help.  Even using the values calculated from the pre-existing code would be fine.

Reply:
There are randomized tests in `inv_proptest()` which should catch issues with other values.

Reply:
There are randomized tests in `inv_proptest()` which should cover all other values.